includes/Modules/Site_Verification.php

```diff
@@ -115,72 +115,90 @@ protected function create_data_request( $method, $datapoint, array $data = array
-				case 'siteverification':
+				case 'verification':
 					if ( ! isset( $data['siteURL'] ) ) {
 						/* translators: %s: Missing parameter name */
 						return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'siteURL' ), array( 'status' => 400 ) );
 					}
+
 					return function() use ( $data ) {
 						$current_user = wp_get_current_user();
+
 						if ( ! $current_user || ! $current_user->exists() ) {
 							return new WP_Error( 'unknown_user', __( 'Unknown user.', 'google-site-kit' ) );
 						}
-						$site = $this->get_data( 'siteverification-list', $data );
+
+						$site = $this->get_data( 'verification', $data );
+
 						if ( is_wp_error( $site ) ) {
 							return $site;
 						}
+
 						$sites = array();
-						if ( isset( $site['verified'] ) && ! $site['verified'] ) {
-							$token = $this->get_data( 'siteverification-token', $data );
+
+						if ( ! empty( $site['verified'] ) ) {
+							return $site;
+						} else {
+							$token = $this->get_data( 'verification-token', $data );
+
 							if ( is_wp_error( $token ) ) {
 								return $token;
 							}
+
 							$this->authentication->verification_tag()->set( $token['token'] );
+
 							$client     = $this->get_client();
 							$orig_defer = $client->shouldDefer();
 							$client->setDefer( false );
-							$urls   = $this->permute_site_url( $data['siteURL'] );
 							$errors = new WP_Error();
-							foreach ( $urls as $url ) {
+
+							foreach ( $this->permute_site_url( $data['siteURL'] ) as $url ) {
 								$site = new \Google_Service_SiteVerification_SiteVerificationWebResourceResourceSite();
 								$site->setType( 'SITE' );
 								$site->setIdentifier( $url );
 								$resource = new \Google_Service_SiteVerification_SiteVerificationWebResourceResource();
 								$resource->setSite( $site );
+
 								try {
-									$sites[] = $this->get_service( 'siteverification' )->webResource->insert( 'META', $resource ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName
+									$sites[] = $this->get_siteverification_service()->webResource->insert( 'META', $resource );
 								} catch ( Google_Service_Exception $e ) {
-									$message = $e->getErrors();
-									if ( isset( $message[0] ) && isset( $message[0]['message'] ) ) {
-										$message = $message[0]['message'];
-									}
+									$messages = wp_list_pluck( $e->getErrors(), 'message' );
+									$message  = array_shift( $messages );
+
 									$errors->add( $e->getCode(), $message, array( 'url' => $url ) );
 								} catch ( Exception $e ) {
 									$errors->add( $e->getCode(), $e->getMessage(), array( 'url' => $url ) );
 								}
 							}
+
 							$client->setDefer( $orig_defer );
+
 							if ( empty( $sites ) ) {
 								return $errors;
 							}
 						}
+
 						$this->authentication->verification()->set( true );
+
+						$verification = $this->get_siteverification_service()->webResource->get( $data['siteURL'] );
```

Comment:
We should check here if this actually came back successfully, with a try-catch. If not, we can for now just grab the first response in the `$sites` array, which is also a `webResource` (see https://developers.google.com/site-verification/v1/webResource/insert#response). Basically `$verification = array_shift( $sites )`.

Reply:
Done!